Trappist/src/Promact.Trappist.Test/TestConduct/TestConductRepositoryTest.cs

```diff
@@ -1,13 +1,83 @@
-﻿using Xunit;
+﻿using Promact.Trappist.Repository.TestConduct;
+using Xunit;
+using Microsoft.Extensions.DependencyInjection;
+using System.Threading.Tasks;
+using Promact.Trappist.DomainModel.Models.TestConduct;
+using Promact.Trappist.Repository.Tests;
+using Moq;
+using Promact.Trappist.Utility.GlobalUtil;
 
 namespace Promact.Trappist.Test.TestConduct
 {
     [Collection("Register Dependency")]
     public class TestConductRepositoryTest : BaseTest
     {
+        #region Private Variables
+        #region Dependencies
+        private readonly ITestConductRepository _testConductRepository;
+        private readonly ITestsRepository _testRepository;
+        private readonly Mock<IGlobalUtil> _globalUtil;
+        #endregion
+        private readonly DomainModel.Models.Test.Test test = new DomainModel.Models.Test.Test();
+        private readonly TestAttendees model = new TestAttendees();
+        #endregion
+
+        #region Constructor
         public TestConductRepositoryTest(Bootstrap bootstrap) : base(bootstrap)
         {
-            
+            _testConductRepository = _scope.ServiceProvider.GetService<ITestConductRepository>();
+            _testRepository = _scope.ServiceProvider.GetService<ITestsRepository>();
+            _globalUtil = _scope.ServiceProvider.GetService<Mock<IGlobalUtil>>();
+            Initialize();
+        }
+        #endregion
+
+        #region Testing Methods
+        /// <summary>
+        /// This test case used to check test attendee successfully register.
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task ValidRegisterTestAttendees()
+        {
+            await CreateTest();
+            Assert.True(await _testConductRepository.RegisterTestAttendees(model, "H0SGXPOwsR"));
+        }
+
+        /// <summary>
+        /// This test case used to check test attendee already exist.
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        public async Task InvalidRegisterTestAttendees()
+        {
+            await CreateTest();
+            await _testConductRepository.RegisterTestAttendees(model, "H0SGXPOwsR");
+            Assert.False(await _testConductRepository.RegisterTestAttendees(model, "H0SGXPOwsR"));
+        }
+
+        /// <summary>
+        /// This method used for creating a test.
+        /// </summary>
+        /// <returns></returns>
+        private async Task CreateTest()
+        {
+            _globalUtil.Setup(x => x.GenerateRandomString(10)).Returns("H0SGXPOwsR");
```

Comment:
I don't  need to generate random string of test link every time that's why I mocked globalutil.